rapidsai/docker
# `docker` repo overhaul

Closes #539 

This PR overhauls the `docker` according to #539.

Multi-arch builds inspired from https://github.com/rapidsai/mambaforge-cuda/pull/31

Note: this PR merges into `branch-23.08-overhaul` instead of `branch-23.08`. `branch-23.08-overhaul` was manually created to remove the vast majority of the repo. Merging this PR into that branch shows more obvious changes instead of a [long list of deleted files](https://github.com/rapidsai/docker/compare/branch-23.08...branch-23.08-overhaul).

To-do:
- [x] Handle uploading nightlies with alpha version tag
- [x] `main` builds should clean up `condarc`
- [x] Test all notebooks (not just cudf)

Follow up PRs:
- Dockerhub/NGC readme updater
- Enable nightly tests
- Remove required `up-to-date` status check
- Merge `branch-23.08-overhaul` into `branch-23.08`
- Add style checking (https://github.com/rapidsai/docker/pull/545#discussion_r1264520507)
